redfish-core/lib/assembly.hpp

```diff
@@ -365,9 +407,106 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            if (!json_util::readJson(req, asyncResp->res, "Assemblies", assemblyData, "Oem", oem)) 
+            {
+                const nlohmann::json& thisJson = *assemblyData;
+
+                BMCWEB_LOG_ERROR << "Property Value Format Error ";
+                messages::propertyValueFormatError(
+                asyncResp->res,
+                thisJson.dump(2, ' ', true, 
+                              nlohmann::json::error_handler_t::replace),
+                "Assemblies");
+                return;
+            }
+
+            if (!oem)
```

Comment:
If this is causing 
```
tippolit@gfwa166:~/Code/openbmc/build/p10bmc$ curl -k -X PATCH -d '{"Assemblies":[{"MemberId" : "1", "LocationIndicatorActive": true}]}' https://root:0penBmc@ever8bmc:18080/redfish/v1/Chassis/chassis/Assembly
{
"[Oem@Message.ExtendedInfo](mailto:Oem@Message.ExtendedInfo)": [
{
"@odata.type": "#Message.v1_1_1.Message",
"Message": "The property Oem is a required property and must be included in the request.",
"MessageArgs": [
"Oem"
],
"MessageId": "Base.1.8.1.PropertyMissing",
"MessageSeverity": "Warning",
"Resolution": "Ensure that the property is in the request body and has a valid value and resubmit the request if the operation failed."
}
]
}
```
Then let's do 
```
if (OEM)
{

            std::optional<nlohmann::json> openbmc;
            if (!json_util::readJson(*oem, asyncResp->res, "OpenBMC", openbmc))
            {
...
```

and remove this LOG_ERROR and propertyMissing